tests/test_security_path_traversal.py

```
@@ -204,3 +204,71 @@ def test_run_tailer_path_validation(tmp_path):
+
+def test_wisdom_api_path_traversal(tmp_path, monkeypatch):
+    """Test that Wisdom API endpoints validate path components against traversal."""
+    import asyncio
+    from fastapi import HTTPException
+    from swarm.api.routes import wisdom
+
+    # Mock to avoid SpecManager initialization errors
+    monkeypatch.setattr(wisdom, "_get_runs_root", lambda: tmp_path)
+    monkeypatch.setattr(wisdom, "_get_repo_root", lambda: tmp_path)
+
+    async def run_tests():
```

Ruff (0.15.4)

[warning] Missing return type annotation for private function `run_tests`

Add return type annotation: `None`

(ANN202)